infrastructure-scripts/Get-SqlDbServiceAccount.ps1

```diff
@@ -0,0 +1,144 @@
+
+.DESCRIPTION
+Create A Sql Database Service Account Automation
+
+.PARAMETER ServerNamr
+Name of the SQL Server
+
+.PARAMETER AzureFirewallName
+Name of the tempoary Sql Server Firewall rule created
+
+.PARAMETER SqlUserName
+Sql Server Master User Name
+
+.PARAMETER SqlPassword
+Sql Server Master Password.
+
+.PARAMETER SqlServiceAccountName
+The name of the service account to be created.
+
+.Parameter SQLResourceGroup
+The Resource group of the SQL Server
+
+.PARAMETER Enviroment
+The Enviroment of the New Service account,
+
+.PARAMETER KeyVaultName
+The name of the Keyvault for the Enviroment
+
+
+.EXAMPLE
+
+.\Get-SqlDbServiceAccount.ps1  -ServerName ServerName `
+                                   -DataBaseName DataBaseName `
+                                   -SqlUserName SqlUserName `
+                                   -SqlPassword SqlPassword `
+                                   -SqlServiceAccountName SqlServiceAccountName `
+                                   -SQLResourceGroup SQLResourceGroup
+                                   -Enviroment Enviroment
+                                   -KeyVaultName KeyVaultName
+
+#>
+
+[CmdletBinding(DefaultParameterSetName = 'None')]
+param
+(
+    [Parameter(Mandatory = $true)]
+    [String] $ServerName,
+    [String] $AzureFirewallName = "AzureWebAppFirewall",
+    [Parameter(Mandatory = $true)]
+    [String]  $DataBaseName,
+    [Parameter(Mandatory = $true)]
+    [String]$SqlUserName,
+    [Parameter(Mandatory = $true)]
+    [SecureString]  $SqlPassword,
+    [Parameter(Mandatory = $true)]
+    [String]  $SqlServiceAccountName,
+    [Parameter(Mandatory = $true)]
+    [String]  $SQLResourceGroup,
+    [Parameter(Mandatory = $true)]
+    [String]  $Enviroment,
+    [Parameter(Mandatory = $true)]
+    [String] $KeyVaultName
+)
+
+$ErrorActionPreference = 'Stop'
+
+
+function New-AzureSQLServerFirewallRule {
+    $agentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
+    New-AzureRmSqlServerFirewallRule -StartIPAddress $agentIp -EndIPAddress $agentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+}
+function Update-AzureSQLServerFirewallRule {
+    $agentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
+    Set-AzureRmSqlServerFirewallRule -StartIPAddress $agentIp -EndIPAddress $agentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+}
+
+function Get-RandomPassword {
+    $Password = ([char[]]([char]33..[char]95) + ([char[]]([char]97..[char]126)) + 0..9 | Sort-Object { Get-Random })[0..33] -join ''
+    return $Password
+}
+function Invoke-SqlQuery([String] $query) {
+    $output = Invoke-Sqlcmd -ServerInstance $ServerFQDN -Database $DataBaseName -Username $SqlUserName -Password $SqlPassword -OutputSqlErrors $true -Query $query
+    return $output
+}
+
+function Remove-AzureSQLServerFirewallRule {
```

Comment:
Questioning the need for functions if the code is only used once? Happy either way with this though.

Reply:
Nicked the code :-)